js/app.js

```diff
@@ -131,392 +58,160 @@ app.controller('CheckoutController', function($scope, $interval, Order, $httpPar
-            else
-                params = {};
-            params.uuid = uuid;
-            url = window.location.protocol + "//" + window.location.host + window.location.pathname;
-            var serializedParams = $httpParamSerializer(params);
-            if (serializedParams.length > 0) {
-                url += ((url.indexOf('?') === -1) ? '?' : '&') + serializedParams;
+            $scope.tick_interval = $interval($scope.tick, 1000);
+            //Connect and Listen on websocket for payment notification
+            var ws = new ReconnectingWebSocket("wss://" + subdomain + ".blockonomics.co/payment/" + $scope.order.address + "?timestamp=" + $scope.order.timestamp);
+            ws.onmessage = function(evt) {
+                ws.close();
+                $interval(function() {
+                    //Redirect to order received page if message from socket
+                    window.location = Url.get_wc_endpoint('finish_order', $scope.order_id);
+                //Wait for 2 seconds for order status to update on server
+                }, 2000, 1);
             }
-            return url;
-        }
-
-    //Increment altcoin timer
-    $scope.alt_tick = function() {
-        $scope.alt_clock = $scope.alt_clock - 1;
-        $scope.alt_progress = Math.floor($scope.alt_clock * totalProgress / alt_totalTime);
-        if ($scope.alt_clock < 0) {
-            $scope.alt_clock = 0;
-            //Altcoin order expired
-            $interval.cancel($scope.alt_tick_interval);
         }
-        $scope.alt_progress = Math.floor($scope.alt_clock * totalProgress / alt_totalTime);
-    };
-
-    //Send altcoin refund email 
-    function send_refund_email() {
-        uuid = get_uuid();
-        WpAjax.get({
-            action: 'send_email',
-            order_id: $scope.order.order_id,
-            order_uuid: uuid,
-            order_coin: $scope.altcoinselect,
-            refund_address: $scope.altrefund
-        });
     }
-
-    //Update the altcoin status
-    function update_altcoin_status(status) {
-        $scope.order.altstatus = status;
+    
+    //Check if the blockonomics order is present
+    function check_blockonomics_order() {
+        $scope.spinner = true;
+        if (typeof $scope.order_id != 'undefined') {
+            //Fetch the order using order_id
+            Order.get({
+                "get_order": $scope.order_id,
+                "crypto": Url.get_parameter_by_name("crypto")
+            }, function(data) {
+                $scope.spinner = false;
+                if(data.address !== undefined){
+                    $scope.order = data;
+                    // show the checkout page
+                    proccess_order_data();
+                    $scope.checkout_panel  = true;
+                }else if(data.crypto.code === 'btc'){
+                    $scope.address_error_btc = true;
+                }else if(data.crypto.code === 'bch'){
+                    $scope.address_error_bch = true;
+                }
```

Comment:
Server return empty data when new_address gen fails. So this will not work. Better to just use *crypto* variable that we already have.

Reply:
Fixed with 7e2bded